### DISCUSSION TITLE
Sync relay status with switch status after a reboot/power loss

### DISCUSSION THREAD

**Discussion by Sekenenz** [arendst/Tasmota#20486] [Support request] [answered]
(beginner user)
I've configured a [board](https://templates.blakadder.com/ESP12F_Relay_X1.html) to use a reed switch as contact sensor on GPIO4 (switch 2) that control a dummy relay on GPIO3 (relay 2). 

<img width="319" alt="Screenshot 2024-01-13 185609" src="https://github.com/arendst/Tasmota/assets/17243492/078bba88-b6a8-4882-8b83-251d56dc3111">

<img width="305" alt="Screenshot 2024-01-13 184109" src="https://github.com/arendst/Tasmota/assets/17243492/7c5eec39-1a01-4c3d-8b1c-3a9113f3e826">

"SwitchMode2" is 1 so it follows the switch2's gpio status, but in the eventuality the contact sensor change status when offline for a reboot or power loss , when back online the sensor is out sync because it remember the last state. "SetOption0" is ON because I need that relay1 remember his state, but how do I set relay2 to sync with switch2 after reboot/powerloss?

**Comment by sfromis** [comment]
Using rules with `Switch2#boot` allows you to capture state at boot, and do something with it, including setting a relay to the matching state.

**Reply by Sekenenz** [reply]
Something like...
ON Switch2#boot DO < ? > ENDON

Can you help me?

**Reply by sfromis** [reply, marked as answer]
Well to turn on/off, that would be the Power command for the relay in question.
That could be
`on Switch2#boot do Power2 %value% endon`

Or if that's the wrong way, `Scale` can be used to flip the value between 0 and 1.
`on Switch2#boot do Scale2 %value%,1,0,0,1 endon`
`on var2#state do Power2 %value% endon`
Better do a `CalcRes 0` once only, to avoid decimals on the calculations.

**Reply by Sekenenz** [reply]
But that would set a value unconditional, right? Isn't there a command to read gpio4 value and set gpio3 accordingly?

**Reply by sfromis** [reply]
`Switch2#boot` is based on current value of the Switch pin, available to the rule as `%value%`

**Reply by Sekenenz** [reply]
ok perfect, thanks
